src/cloud.ts

```diff
@@ -0,0 +1,479 @@
+// @ts-nocheck
+/**
+ * TEMPORARY workaround:
+ * Should re-write this in typescript in the future
+ *
+ * Implements the wordcloud algorithm based on: https://github.com/jasondavies/d3-cloud/tree/v1.2.5
+ * Improvements:
+ *  Use a setTimeout + batch sizes to calculate clouds in order to avoid blocking main thread for two long (87-101)
+ *
+ */
+
+// Word cloud layout by Jason Davies, https://www.jasondavies.com/wordcloud/
+// Algorithm due to Jonathan Feinberg, http://static.mrfeinberg.com/bv_ch03.pdf
+import { Cloud } from './types';
+var dispatch = require('d3-dispatch').dispatch;
+
+var cloudRadians = Math.PI / 180,
+  cw = (1 << 11) >> 5,
+  ch = 1 << 11;
+
+export default function<T>(): Cloud<T> {
+  var size = [256, 256],
+    text = cloudText,
+    font = cloudFont,
+    fontSize = cloudFontSize,
+    fontStyle = cloudFontNormal,
+    fontWeight = cloudFontNormal,
+    rotate = cloudRotate,
+    padding = cloudPadding,
+    spiral = archimedeanSpiral,
+    words = [],
+    timeInterval = Infinity,
+    event = dispatch('word', 'end'),
+    timer = null,
+    random = Math.random,
+    cloud = {},
+    canvas = cloudCanvas;
+
+  let killed = false;
+
+  cloud.canvas = function(_) {
+    return arguments.length ? ((canvas = functor(_)), cloud) : canvas;
+  };
+
+  cloud.start = function() {
+    var contextAndRatio = getContext(canvas()),
+      board = zeroArray((size[0] >> 5) * size[1]),
+      bounds = null,
+      n = words.length,
+      i = -1,
+      tags = [],
+      data = words
+        .map(function(d, i) {
+          d.text = text.call(this, d, i);
+          d.font = font.call(this, d, i);
+          d.style = fontStyle.call(this, d, i);
+          d.weight = fontWeight.call(this, d, i);
+          d.rotate = rotate.call(this, d, i);
+          d.size = ~~fontSize.call(this, d, i);
+          d.padding = padding.call(this, d, i);
+          return d;
+        })
+        .sort(function(a, b) {
+          return b.size - a.size;
+        });
+
+    // Added by react-wordcloud
+    // Allows to calculate a subset of data instead of all of the words at once
+    function multiStep(from: number, to: number) {
+      for (let i = from; i < to; i += 1) {
+        var d = data[i];
+        d.x = (size[0] * (random() + 0.5)) >> 1;
+        d.y = (size[1] * (random() + 0.5)) >> 1;
+        cloudSprite(contextAndRatio, d, data, i);
+        if (d.hasText && place(board, d, bounds)) {
+          tags.push(d);
+          event.call('word', cloud, d);
+          if (bounds) cloudBounds(bounds, d);
+          else
+            bounds = [
+              { x: d.x + d.x0, y: d.y + d.y0 },
+              { x: d.x + d.x1, y: d.y + d.y1 },
+            ];
+          // Temporary hack
+          d.x -= size[0] >> 1;
+          d.y -= size[1] >> 1;
+        }
+      }
+    }
+
+    // Added by react-wordcloud
+    // Iterates dataset using setTimeout in order to prevent blocking of the main thread
+    function loop(i: number) {
```

Comment:
Going through the PR to test the code and behaviors.  This function is defined and clearly used, but I am unable to trace it in the code (no mention of calling `loop` anywhere in our new code or in `d3-cloud`).

Any idea where this is being called?  I'll submit an update to this PR on some cleanup.